snowflake_data_profiler/profiling/profiler.py

```diff
@@ -50,21 +55,25 @@ def connect_to_snowflake(sfUser, sfPswd, sfURL, sfDatabase, sfSchema, sfTable, s
     return df
 
 def get_profile_results(data):
-    profile = ProfileReport(
-      data, 
-      title='Snowflake Data Profiler from Hashmap', 
-      progress_bar=False, 
-      explorative=True, 
-      correlations={
-         "pearson": {"calculate": True},
-         "spearman": {"calculate": False},
-         "kendall": {"calculate": False},
-         "phi_k": {"calculate": False},
-         "cramers": {"calculate": False},
-     },
-    )
-    p = profile.to_html() # this step sometimes fails with matplotlib errors about threads. I've only fixed it by adjusting requirements.txt in the past. I've just specified the specific versions of libraries. Pyarrow seems to have an impact on this.
-    return p
+    if isinstance(data, pd.DataFrame):
```

Comment:
There are some more advanced ways to do this kind of type-strict checking.

Check out the following about this:
- https://medium.com/@ageitgey/learn-how-to-use-static-type-checking-in-python-3-6-in-10-minutes-12c86d72677b
- https://docs.python.org/3/library/typing.html

Many backend devs prefer stronger typing. It's good that you're thinking this way.